androbot/handlers.py

```diff
@@ -391,10 +394,44 @@ async def send_correct_answer_to_user(message: aiotypes.Message):
         reply_markup=view.markup,
     )
 
-    await DialogueStates.GOT_ANSWER.set()
+    if view.state:
+        await view.state.set()
 
 
-@dp.message_handler(regexp="Отправь материалы", state=DialogueStates.GOT_ANSWER)
+@dp.message_handler(regexp="Верный", state=DialogueStates.GOT_ANSWER)
+@dp.message_handler(regexp="Частично верный", state=DialogueStates.GOT_ANSWER)
+@dp.message_handler(regexp="Неверный", state=DialogueStates.GOT_ANSWER)
```

Comment:
Вообще, давай это как-то переделаем
Например в types_.user_score написать такое:

```python
i2score = [
"Верный",
"Частично верный",
"Неверный",
]
score2i = {s: i for (i, s) in enumerate(i2score)}
```

Reply:
Тогда можно будет так сделать:
```suggestion
@dp.message_handler(regexp="|".join(i2score), state=DialogueStates.GOT_ANSWER)
```

Reply:
хорошо, спасибо за идею

Reply:
думаю сделать прямо в UserScore описание

Reply:
В одном бы месте как-то хранить всю эту информацию

Reply:
я вот так сделал, еще пока не протестировал, завтра пушну, но вроде должно работать.

`class UserScore(Enum):
    WRONG = 0, "Неверный"
    PARTLY = 1, "Частично верный"
    RIGHT = 2, "Верный"

    def __new__(cls, *args, **kwargs):
        obj = object.__new__(cls)
        obj._value_ = args[0]
        return obj

    def __init__(self, _: str, description: str = None):
        self._description_ = description

    @property
    def description(self):
        return self._description_

    @classmethod
    def by_description(cls, find_description: str) -> Optional[Enum]:
        matched = [x for x in UserScore if x.description.lower() in find_description.lower()]
        if matched:
            return matched[0]
        return None`

Reply:
Использование:
```
@dp.message_handler(regexp="|".join(score.description for score in UserScore), state=DialogueStates.GOT_ANSWER)
```
```
answer_score = UserScore.by_description(message.text)
```

Reply:
выглядит виртуозно, конечно

Reply:
Если често, код в интернете взял. Сам бы так не написал. 😉